kafka-connect-mtls/check-ssl-client-auth.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+verify_ssl_client_auth () {
+
+  openssl pkcs12 -in connect/secrets/$1.p12 -out connect/secrets/ca.pem -cacerts -nokeys -passin pass:confluent -passout pass:confluent
+  openssl pkcs12 -in connect/secrets/$1.p12 -out connect/secrets/client.pem -clcerts -nokeys -passin pass:confluent -passout pass:confluent
+  openssl pkcs12 -in connect/secrets/$1.p12 -out connect/secrets/key.pem -nocerts -passin pass:confluent -passout pass:confluent
+
+  curl --insecure --key connect/secrets/key.pem --cacert connect/secrets/ca.pem --cert connect/secrets/client.pem:confluent https://localhost:18083
```

Comment:
--insecure is not needed here (or --cacert)

Reply:
You do not necessary need pem-s here, you could use pksc12 with curl, e.g.
```
curl --cacert connect/secrets/ca.pem --cert-type PKSC12 --cert connect/secrets/client.p12:confluent https://localhost:18083
```
OR 
```
# as CA testing is not a case here
curl --insecure --cert-type PKSC12 --cert connect/secrets/client.p12:confluent https://localhost:18083
```

Reply:
good catch ;-)

Reply:
Hi @kikulikov, 
  I tried the --insecure and --cacert, but as far as I understand curl needs it, otherwise for example my _curl_ version barf an error:

```
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
HTTPS-proxy has similar options --proxy-cacert and --proxy-insecure.
```

that make a lot of sense, because the CA that i use here, is not in the "bundle".

agree?

Reply:
regarding the cert type, are you talking about linux curl vs mac curl ? at less the version I do have in my machine only accepts 
``
--cert-type TYPE  Certificate file type (DER/PEM/ENG) (SSL)
```

Reply:
context, this is the curl I do have in my system:

```
curl 7.54.0 (x86_64-apple-darwin17.0) libcurl/7.54.0 LibreSSL/2.0.20 zlib/1.2.11 nghttp2/1.24.0
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy 
```

Reply:
Are you on linux?

Reply:
So it should be `curl --insecure --cert-type P12 ....`

Reply:
But I guess ignore it... if it is not common and behaves quite differently on various env

Reply:
@purbon I'm not sure I understood your comment `I tried the --insecure and --cacert, b .....`

Reply:
updated with proper cert structure, so insecure is not raised anymore.